### PR TITLE
feat: [CDS-38405]: service override as an entity

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
##### Summary:

Add new tab to handle service override as an entity

##### Jira Links:

https://harness.atlassian.net/browse/CDS-38405

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/EnvironmentConfiguration.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      value: 'Production'
+    },
+    {
+      label: getString('cd.preProduction'),
+      value: 'PreProduction'
+    }
+  ]
+
+  const { data: environmentSchema } = useGetYamlSchema({
+    queryParams: {
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/EnvironmentConfiguration.tsx:89]: Please check if this is updated schema. We had new api for service entity schema

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/EnvironmentConfiguration.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            <Text
+              color={Color.GREY_450}
+              margin={{ top: 'medium', bottom: 'small' }}
+              font={{ variation: FontVariation.FORM_LABEL, weight: 'bold' }}
+            >
+              {getString('envType')}
+            </Text>
+            <ThumbnailSelect className={css.thumbnailSelect} name={'type'} items={typeList} />
+          </Card>
+          {/* #region Advanced section */}
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/EnvironmentConfiguration.tsx:149]: Remember to remove this if not needed

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+  const { getString } = useStrings()
+  const { showSuccess, showError } = useToaster()
+  const { getRBACErrorMessage } = useRBACError()
+  const { expressions } = useVariablesExpression()
+  const [selectedVariable, setSelectedVariable] = useState<VariableState | null>(null)
+  const [isEdit, setIsEdit] = useState<boolean>(false)
+
+  const { data: services, loading: servicesLoading } = useGetServiceList({
+    queryParams: {
+      accountIdentifier: accountId,
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx:68]: We need to pass new `type` query param. You can pass that after https://github.com/harness/harness-core-ui/pull/9314 merge

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+      projectIdentifier
+    }
+  })
+
+  const {
+    data: serviceOverrides,
+    loading: serviceOverridesLoading,
+    refetch: refetchServiceOverrides
+  } = useGetServiceOverridesList({
+    queryParams: {
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx:79]: Check if we need deployment type here as well

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/AddEditServiceOverride.tsx [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+            }),
+            []
+          )
+        )
+      }
+    }
+  }, [])
+
+  const { data: environmentGroupSchema } = useGetYamlSchema({
+    queryParams: {
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/AddEditServiceOverride.tsx:150]: Check for yamlschema api

**Review by vkbansal — APPROVED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/EnvironmentConfiguration.tsx (hunk 2 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/EnvironmentConfiguration.tsx:149]: This is a comment for folding code. I can remove it

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/EnvironmentConfiguration.tsx (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/EnvironmentConfiguration/EnvironmentConfiguration.tsx:89]: This is updated schema

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx (hunk 3 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceOverrides.tsx:68]: Not required for this. We will need to fetch all service overrides as one environment is not linked to a deployment type

**Review by bhavya-sinha — DISMISSED** [review]

**Review by vkbansal — DISMISSED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Review by vkbansal — APPROVED** [review]

**Pull request merged**